pss/pss_test.go

```diff
@@ -70,6 +71,117 @@ func TestTrojanChunkRetrieval(t *testing.T) {
 
 }
 
+func TestPssMonitor(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+
+	db, cleanupFunc := newTestDB(t, &Options{Tags: chunk.NewTags()})
+	defer cleanupFunc()
+
+	testTargets := [][]byte{
+		{57, 120},
+		{209, 156},
+		{156, 38},
+		{89, 19},
+		{22, 129}}
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY")
+
+	var ch chunk.Chunk
+
+	pss := NewPss(db.localStore)
+
+	// call Send to store trojan chunk in localstore
+	if ch, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+	item, err := db.localStore.pullIndex.Get(shed.Item{
+		Address: ch.Address(),
+		BinID:   1,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = db.localStore.Set(context.Background(), chunk.ModeSetSyncPull, ch.Address())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	checkTag(t, tag, 0, 1, 0, 1, 0, 1)
+
+}
+
+// CheckTag checks the first tag in the api struct to be in a certain state
+// TODO: reuse existing tag CheckTag instead of this
+func checkTag(t *testing.T, tag *chunk.Tag, split, stored, seen, sent, synced, total int64) {
+	t.Helper()
+	if tag == nil {
+		t.Fatal("no tag found")
+	}
+	tSplit := tag.Get(chunk.StateSplit)
+	if tSplit != split {
+		t.Fatalf("should have had split chunks, got %d want %d", tSplit, split)
+	}
+
+	tSeen := tag.Get(chunk.StateSeen)
+	if tSeen != seen {
+		t.Fatalf("should have had seen chunks, got %d want %d", tSeen, seen)
+	}
+
+	tStored := tag.Get(chunk.StateStored)
+	if tStored != stored {
+		t.Fatalf("mismatch stored chunks, got %d want %d", tStored, stored)
+	}
+
+	tSent := tag.Get(chunk.StateSent)
+	if tStored != stored {
+		t.Fatalf("mismatch sent chunks, got %d want %d", tSent, sent)
+	}
+
+	tSynced := tag.Get(chunk.StateSynced)
+	if tSynced != synced {
+		t.Fatalf("mismatch synced chunks, got %d want %d", tSynced, synced)
+	}
+
+	tTotal := tag.TotalCounter()
+	if tTotal != total {
+		t.Fatalf("mismatch total chunks, got %d want %d", tTotal, total)
+	}
+}
+
+// newTestDB is a helper function that constructs a
+// temporary database and returns a cleanup function that must
+// be called to remove the data.
+// TODO: refactor into common newTestDB for all test
+func newTestDB(t testing.TB, o *Options) (db *DB, cleanupFunc func()) {
```

Comment:
could you please explain what the difference is (conceptually) between this func and `newMockLocalStore`?

Reply:
I was trying to get *DB to use the pullIndex, one exposed only localstore.
I was unable to get pullIndex due to fields exposed in the localstore package.
I currently removed this